### PR TITLE
Add checking for PHI within deidentified Patient and Coverage Resources.

### PR DESCRIPTION
**Pull request by plarocque4**
Test this using the crd-request-generator changes within the phi branch.

### CODE REVIEW

File: server/src/main/java/org/hl7/davinci/endpoint/components/CardBuilder.java [hunk 1]
```diff
@@ -396,6 +401,10 @@ private static Card baseCard(CardTypes cardType, String requestId) {
       cardExtension.addAssociatedResource(requestId);
       card.setExtension(cardExtension);
     }
+
+    if (deidentifiedResourcesContainPhi) {
+      card.setDetail("Note: de-identified resources provided in request contain Protected Health Information (PHI). Please notify administrator.");
```
Comment (yuliumitre) [inline server/src/main/java/org/hl7/davinci/endpoint/components/CardBuilder.java:406]: I guess this is a warning from the system. Is this something the end user should know?

Comment (plarocque4) [reply server/src/main/java/org/hl7/davinci/endpoint/components/CardBuilder.java:406]: You are correct. I posed this question to Jeff but he wasn't sure where the warning/error should go. For now I put the message in with the thought that I can easily remove it later if needed.

**Review by yuliumitre — APPROVED** [review]
It works as expected. The PHI fields are not passed in for Patient and Coverage when the option is enabled.

File: server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/r4/FhirBundleProcessor.java [hunk 2]
```diff
@@ -38,10 +43,96 @@ public FhirBundleProcessor(FileStore fileStore, String baseUrl) {
...
+          }
+          
+          // check the birthdate
+          Date now = new Date();
+          long diffInMs = Math.abs(now.getTime() - patient.getBirthDate().getTime());
+          long diffInDays = TimeUnit.DAYS.convert(diffInMs, TimeUnit.MILLISECONDS);
+          String birthDateStr = patient.getBirthDateElement().asStringValue();
+
+          // if age is less than 2 years then there should be a year and month
+          if (diffInDays < (265 * 2)) {
```
[the remark this answers is not in the document]
Comment (plarocque4) [reply server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/r4/FhirBundleProcessor.java:89]: Haha, must have been late at night that I implemented this. At least that's the excuse I'm sticking with.

**Review by RobertScalfani — APPROVED** [review]

**Pull request merged**